compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -622,10 +622,12 @@ impl Diagnostic {
     pub fn multipart_suggestion_with_style(
         &mut self,
         msg: impl Into<SubdiagnosticMessage>,
-        suggestion: Vec<(Span, String)>,
+        mut suggestion: Vec<(Span, String)>,
         applicability: Applicability,
         style: SuggestionStyle,
     ) -> &mut Self {
+        suggestion.dedup();
+
```

Comment:
Should we dedup after sorting? In case we have `[(first span, first stuff), (second span, second stuff), (first span, first stuff)]`.

Reply:
Great tips! Before this, I always assumed that ﻿`Vec::dedup` was equivalent to ﻿`Vec::from(HashSet::from(vector))`.